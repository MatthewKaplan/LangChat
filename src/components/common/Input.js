import React from 'react';
import { TextInput, View, Text } from 'react-native';

const Input = ({ label, value, onChangeText, placeholder, secureTextEntry }) => {
	const { inputStyle, labelStyle, containerStyle } = styles;

	return (
		<View style={containerStyle}>
			<Text style={labelStyle}>{label}</Text>
			<TextInput
				secureTextEntry={secureTextEntry}
				autoCorrect={false}
				value={value}
				onChangeText={onChangeText}
				style={inputStyle}
				placeholder={placeholder}
			/>
		</View>
	);
};

const styles = {
	inputStyle: {
		color: '#000',
		paddingLeft: 5,
		paddingRight: 5,
		fontSize: 18,
		lineHeight: 23,
		width: '100%'
	},
	labelStyle: {
		fontSize: 18,
		paddingLeft: 10,
		paddingRight: 10
	},
	containerStyle: {
		height: 40,
		flexDirection: 'row',
		alignItems: 'center'
	}
};

export { Input };
