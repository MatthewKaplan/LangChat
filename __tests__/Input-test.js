// __tests__/Input-test.js
import React from 'react';
import Input from '../src/components/common/Input';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
	const tree = renderer.create(<Input />).toJSON();
	expect(tree).toMatchSnapshot();
});
