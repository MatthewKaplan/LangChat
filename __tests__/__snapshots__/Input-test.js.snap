// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 50,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000",
        "fontSize": 18,
        "height": "100%",
        "lineHeight": 23,
        "margin": "auto",
        "paddingLeft": 5,
        "paddingRight": 5,
        "width": "95%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
